frontend/ui/widget/qrwidget.lua

```diff
@@ -27,17 +27,19 @@ function QRWidget:init()
         return
     else
         local scale
-        if self.width then
+        if self.scale_factor then
+            scale = self.scale_factor
```

Comment:
I don't see the relation between scale (badly named, I take it is the individual square_size, right?) and self.scale_factor (the one used by ImageWidget)

Reply:
OK, I was confusing two problems (scale and square size: I thought I had to care about scaling "manually", and I saw square size as the mean to get it).